.github/workflows/cypress-tests-runner.yml

```
@@ -0,0 +1,216 @@
+name: Run Cypress tests
+
+on:
+  merge_group:
+    types:
+      - checks_requested
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  CONNECTORS: stripe
+  RUST_BACKTRACE: short
+  RUSTUP_MAX_RETRIES: 10
+  SKIP_STEP: ${{ !((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) || !(github.event_name == 'merge_group')}}
```

This condition is incorrect:

The condition to run tests:
```suggestion
  RUN_TESTS: ${{ ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) || (github.event_name == 'merge_group')}}
```

The condition to skip tests:
```suggestion
  SKIP_TESTS: ${{ !((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) && !(github.event_name == 'merge_group')}}

I'd personally prefer using the first one, having the condition to run tests rather than skipping.